wallet-test/src/test/scala/org/bitcoins/wallet/UTXOHandlingTest.scala

```diff
@@ -0,0 +1,57 @@
+package org.bitcoins.wallet
+
+import org.bitcoins.core.protocol.script.EmptyScriptPubKey
+import org.bitcoins.core.wallet.utxo.TxoState._
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.bitcoins.testkit.wallet.WalletTestUtil._
+import org.scalatest.FutureOutcome
+
+class UTXOHandlingTest extends BitcoinSWalletTest {
```

Comment:
other cases that aren't tested AFAICT

1. Coinbase utxos
2. CPFP utxos
3. RBF
4. Reorg scenarios (#2179)
5. marking a utxo as reserved -> unreserved

Reply:
These are handled in `UTXOLifeCycleTest`, `WalletSendingTest`, and `WalletIntegrationTest`